src/controllers/legacycontrollermapping.h

```diff
@@ -1,14 +1,24 @@
 #pragma once
 
+#include <util/assert.h>
+
 #include <QDebug>
 #include <QDir>
+#include <QDomElement>
 #include <QHash>
 #include <QList>
 #include <QSharedPointer>
 #include <QString>
+#include <QtGlobal>
 #include <memory>
+#include <vector>
 
+#include "controllers/legacycontrollersettings.h"
+#include "controllers/legacycontrollersettingslayout.h"
 #include "defs_urls.h"
+#include "preferences/usersettings.h"
+
+#define CONTROLLER_SETTINGS_PREFERENCE_GROUP_KEY "[ControllerSettings_%1_%2]"
```

Comment:
Is it okay to keep it as it? Or shall we generate a UUID from that pair?

Reply:
How does this work when there are two identical controllers connected that might want different settings? This would be common for setups with CDJs or other modular setups (Behringer CMD series for instance). Since you substitute in the controller-name, do we have stability guarantees (what settings get assigned to what piece of hardware, etc?).

Also in terms of codestyle, a QStringLiteral in the .cpp probably makes more sense than a #define in the header.

Reply:
How does that currently work? I have used this as key as it was already used for status and selected mapping.
It terms of guarantees, I think we would need someone to test this. I can try to do so with 2 UHID s but can't guarantee it will be as good as actually trying with two devices.

Reply:
I think your code gets the devicename (it's more than the pure name) from
https://github.com/mixxxdj/mixxx/blob/f391d795b55c460ff0efe34af4f365d1a4dd23b3/src/controllers/hid/hiddevice.cpp#L79-L95
and use it as identifier. I think this should work in multi CDJ setups.

Reply:
Yes, it should, same than how the device name gets fetch for the existing settings AFAIK